### PR TITLE
fix group schema.  fixes #219

### LINKED ISSUE #219: 'dcos marathon groups add' error when dependencies defined
I'm testing marathon's application groups via dcos and created a sample json. When I'm posting it via curl to Marathons REST api, it is successfully started. However via dcos cli I get errors.

If I add a toplevel empty
```"apps": [ ]```
property into the json, then I get:
```RefResolutionError: Unresolvable JSON pointer: u'definitions/pathType'```

Without the ```apps``` property just a lengthy stacktrace.

This is the command I'm using:
```dcos marathon group add multi.json```

The json:
```
{
  "id": "/multi-test7",
  "groups": [ {
    "id": "/multi-test7/persistence",
    "apps": [ {
      "id": "/multi-test7/persistence/mongo",
      "cmd": "while [ true ] ; do echo 'Fake MongoDB server' ; sleep 5 ; done",
      "cpus": 0.1,
      "mem": 64.0,
      "instances": 1
    }]
  },{
    "id": "/multi-test7/persistence2",
    "apps": [ {
      "id": "/multi-test7/persistence2/mongo",
      "dependencies": ["/multi-test7/persistence/mongo"],
      "cmd": "while [ true ] ; do echo 'Fake MongoDB server' ; sleep 5 ; done",
      "cpus": 0.1,
      "mem": 64.0,
      "instances": 1
    }]
  } ]
}
```
After some testing it turned out that if I remove the ```dependencies``` property from the json, it will be posted successfully to Marathon via dcos cli.

### PR DESCRIPTION
**Pull request by mgummelt**
fixes #219

### CODE REVIEW

**Comment by mgummelt** [comment]
@ConnorDoyle plz review

**Comment by ConnorDoyle** [comment]
Thanks @mgummelt.  Please add a test case to exercise this part of the schema (group with dependencies).

**Comment by mgummelt** [comment]
@ConnorDoyle done

File: cli/tests/data/marathon/apps/zero_instance_sleep.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "cpus": 0.1,
   "mem": 16,
   "instances": 0,
+  "dependencies": [],
```
Comment (ConnorDoyle) [inline cli/tests/data/marathon/apps/zero_instance_sleep.json]: In order to test the item schema, we should add at least one dependency.

**Comment by mgummelt** [comment]
I'll go ahead and add one, but I'm not really interested in testing every aspect of this schema.  that would be a lot of tests.

File: cli/tests/data/marathon/apps/zero_instance_sleep.json (hunk 1 above)
Comment (mgummelt) [inline cli/tests/data/marathon/apps/zero_instance_sleep.json]: I'll go ahead and add one, but I'm not really interested in testing every aspect of this schema. that would be a lot of tests.

**Comment by ConnorDoyle** [comment]
Sure, it's just that without it the new dependency field doesn't exercise the change you made to the schema.

**Comment by ConnorDoyle** [comment]
I think adding them iteratively like this as anti-regression measure makes sense.

**Comment by ConnorDoyle** [comment]
Please :ship: once tests are green with the dependency change.

**Comment by mgummelt** [comment]
Actually it looks as if our marathon schema logic is a bit more broken than I thought.  I assigned @tamarrow a JIRA

**Comment by mgummelt** [comment]
Closing for now

**Pull request closed without merging**